sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -53,16 +51,59 @@
     c.client_net_address as client_address,
     sess.host_name as host_name,
     r.*
-FROM sys.dm_tran_active_transactions at
-    INNER JOIN sys.dm_tran_session_transactions st ON st.transaction_id = at.transaction_id
-      LEFT OUTER JOIN sys.dm_exec_sessions sess ON st.session_id = sess.session_id
-    LEFT OUTER JOIN sys.dm_exec_connections c
+FROM sys.dm_exec_sessions sess
+    INNER JOIN sys.dm_exec_connections c
         ON sess.session_id = c.session_id
-    LEFT OUTER JOIN sys.dm_exec_requests r
+    INNER JOIN sys.dm_exec_requests r
         ON c.connection_id = r.connection_id
-        CROSS APPLY sys.dm_exec_sql_text(c.most_recent_sql_handle) text
-    WHERE sess.session_id != @@spid
-    ORDER BY at.transaction_begin_time ASC
+    OUTER APPLY sys.dm_exec_sql_text(c.most_recent_sql_handle) text
+WHERE sess.session_id != @@spid
+ORDER BY r.start_time ASC
```

Comment:
because we truncate based on the size like this https://github.com/DataDog/integrations-core/blob/master/sqlserver/datadog_checks/sqlserver/activity.py#L149-L163

so as soon we we hit the max payload limit we bail, so doing the `ORDER BY` we don't have to sort the rows

Reply:
Oh I see why. It's for applying a limit on the payload size. IMO this sorting should not live with the SQL; it should go with the application logic which applies the limit.